### DISCUSSION TITLE
attach usb device without usbipd tool

### DISCUSSION THREAD

**Discussion by willzyx-hub** [dorssel/usbipd-win#216] [General]
Hello sorry it kinda off topic here xD but just wanna to ask if there is a way to connect usb device to wsl without installing usbipd tool or without typing the command "usbipd wsl attach -b [bus id]" ?

**Comment by dorssel** [comment]
@willzyx-hub 
You will always have to install some tool to expose the USB device, for example `usbipd-win`. So "without installing" ... no.
Or do you mean "auto-attach", without *running* the tool manually? That is already a feature request, see #56.

That last one you can script from Linux, see https://github.com/dorssel/usbipd-win/discussions/168#discussioncomment-1841785.

**Reply by willzyx-hub** [reply]
Yes auto attach without running tool 🙂
